xoinvader/collision.py

```diff
@@ -1,49 +1,188 @@
+        oy2 = abs(tl2.y - tl_ovlp.y)
+        # iterate over overlapping region
+        # and search for collision
+        for i in range(rb_ovlp.x - tl_ovlp.x):
+            for j in range(rb_ovlp.y - tl_ovlp.y):
+                # TODO: check length of current y-level string
+                # it might be not enough to contain i + ox1/2 element
+                if (col1.map[j + oy1][i + ox1] ==
+                        col2.map[j + oy2][i + ox2] == "#"):
+                    return (tl_ovlp, rb_ovlp)
+
+    def _add_collision(self, collider, other_type, callback=None):
+        if collider not in self._colliders:
+            raise ValueError("Attempt to add collision handler for "
+                             "unregistered collider {0}".format(collider))
+        type_pair = TypePair(collider.type, other_type)
+        if type_pair not in self._collisions.keys():
+            self._collisions[type_pair] = [callback]
+        else:
+            self._collisions[type_pair].append(callback)
+
+
+class Collider(object):
+    """Collider component class.
+
+    When added to object, enables it to participate in coliision processing
+    system: i.e. to be able to detect and process collisions between the object
+    and other ones.
+
+    :param str type: name of the collider type. Used in processing possible
+    collisions
+    :param list map: list of strings representing collider geometry. Must be of
+    equal length. "#" character represents actual geometry, all other chars are
+    treated as void space and may be any.
+    :param pos: left top corner of collider map
+    :type pos: :class:`Point`
+    """
+
+    __manager__ = None
+
+    def __init__(self, type, map, pos):
+        if Collider.__manager__ is None:
+            raise ValueError("You can't use Collider objects without "
+                             "ColliderManager. Please create it first.")
+        self._mgr = Collider.__manager__() # manager is weakreaf, thus '()'
+        self._type = type
+        self._pos = pos
+        self._map = map
+        self._mgr._add(self)
+
+    @property
+    def map(self):
+        return self._map
+
+    @property
+    def type(self):
+        return self._type
+
+    @property
+    def pos(self):
+        return self._pos
+
+    def add_handler(self, other_type, callback=None):
+        """Install handler for collision with type `other_type`
+
+        Handler is fired when collision between current collider and any other
+        collider of type `other_type` detected by `update` method of class
+        :class:`CollisionManager`. Callback is `None` by default.
+
+        :param str other_type: type of collider to install collision handler
+        with
+        :param function callback: function which is called when current
+        CollisionManager's update method detects collision of this collider
+        with collider of type `other_type`
+        """
+
+        self._mgr._add_collision(self, other_type, callback)
```

Comment:
No, it should not. `Collider.add_handler` is intended to be used for this purpose.